subs: Migrate subscriptions modal to use simplebar scrollbars.

synicalsyntax opened the pull request:
This PR is a part of solving #10055 and migrates scrollbars only in the subscriptions modal to use simplebar instead of perfect-scrollbar.

Various areas of the modal DOM needed to be refactored to support simplebar.

showell commented:
@synicalsyntax Thanks for working on this!  Looks like there's a merge conflict.  I agree with the overall approach here of introducing new divs for scroll containers.  I recently did the same thing with the buddy list.  It sucks to have deeply nested HTML, but the extra divs give a nice separation of concerns.  All the geometry happens with the outer div, basically, while all the actual app stuff happens with the inner div.

We should chat on #frontend about test-deploying this.  Also, for the commit where you move the buttons, is that theoretically independent from the scrollbar decision?  (just thinking whether we need to squash them after the first round of review)

synicalsyntax commented:
@showell I fixed the merge conflict, but I realized by moving to simplebar, we'd break the recently-merged scrolling in the streams list that I implemented in #10010. I'm currently looking into refactoring to match simplebar.

About streams creation: Since the sticky modal footer was being included in the scrollable area, the scrollbar would overlap over the sticky footer, so I refactored a bunch of CSS/JS to move it out of the scrollable area. It's merely a small UI concern, so can be dropped if you believe it's negligible.

showell commented:
For the commit related to the buttons, my only concern was whether it works as a standalone commit or was somehow tied to the scrollbar stuff.  We should merge it if it's an improvement.

If this commit requires stuff from #10010 to work, perhaps we should just have a single PR?

synicalsyntax commented:
@showell No, it doesn't require #10010 to work, but I'm just noting that if you noticed something weird while testing scrolling the streams list with the PR, it's because #10010's remaining commits were never merged.

timabbott commented:
So, I test-deployed this on chat.zulip.org, and there's definitely problems; after doing a bit of fiddling (swapping panes, subscribe/unsubscribe), I was able to get the scrollbar to go off the bottom of the screen.  I then reloading the page, and found the scrollbar just goes off the bottom of the screen by default.  I wonder if there's an interaction with `list_render` here that's at issue?

synicalsyntax commented:
@timabbott Can you show a GIF of what you're seeing? Someone reported on czo about scrolling issues as well, but I can't reproduce it. ([conversation](https://chat.zulip.org/#narrow/stream/9-issues/topic/streams.20settings.20odd))

showell commented:
We don’t use list_render in the stream settings left panel, if that’s the thing we are discussing.
